pkgs/development/python-modules/fastparquet/default.nix

```diff
@@ -15,18 +19,38 @@
 
 buildPythonPackage rec {
   pname = "fastparquet";
-  version = "0.8.1";
-  format = "setuptools";
+  version = "2022.12.0";
+  format = "pyproject";
 
   disabled = pythonOlder "3.7";
 
   src = fetchFromGitHub {
     owner = "dask";
     repo = pname;
     rev = version;
-    hash = "sha256-rWrbHHcJMahaUV8+YuKkZUhdboNFUK9btjvdg74lCxc=";
+    leaveDotGit = true;
+    hash = "sha256-Ya9gLQXTOtGTY3+AyF7Ls01a3r7R9SESq2qDRFjrd5M=";
   };
 
+  nativeBuildInputs = [
+    cython
+    gitMinimal
```

Comment:
Is this instead of `SETUPTOOLS_SCM_PRETEND_VERSION = version;`? The `leaveDotGit = true;` has serious issues with reproducibility.

Reply:
Instead of using versioneer. The gitMinimal dependency I forgot to drop.

Reply:
FWIW: This is how it misbehaves without git. Looking to patch that away.

```
fastparquet> Executing pipBuildPhase
fastparquet> Creating a wheel...
fastparquet> WARNING: The directory '/homeless-shelter/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you should use sudo's -H flag.
fastparquet> Processing /build/source
fastparquet>   Running command Preparing metadata (pyproject.toml)
fastparquet>   Compiling fastparquet/speedups.pyx because it changed.
fastparquet>   Compiling fastparquet/cencoding.pyx because it changed.
fastparquet>   [1/2] Cythonizing fastparquet/cencoding.pyx
fastparquet>   [2/2] Cythonizing fastparquet/speedups.pyx
fastparquet>   Traceback (most recent call last):
fastparquet>     File "/nix/store/bk06jpp1hswx6i8zyzbk4lgkfyq92qic-python3.10-pip-22.3.1/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 351, in <module>
fastparquet>       main()
fastparquet>     File "/nix/store/bk06jpp1hswx6i8zyzbk4lgkfyq92qic-python3.10-pip-22.3.1/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 333, in main
fastparquet>       json_out['return_val'] = hook(**hook_input['kwargs'])
fastparquet>     File "/nix/store/bk06jpp1hswx6i8zyzbk4lgkfyq92qic-python3.10-pip-22.3.1/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 152, in prepare_metadata_for_build_wheel
fastparquet>       return hook(metadata_directory, config_settings)
fastparquet>     File "/nix/store/0d7f64w6zqxvn411gas3bqvpcik84x82-python3.10-setuptools-65.6.3/lib/python3.10/site-packages/setuptools/build_meta.py", line 377, in prepare_metadata_for_build_wheel
fastparquet>       self.run_setup()
fastparquet>     File "/nix/store/0d7f64w6zqxvn411gas3bqvpcik84x82-python3.10-setuptools-65.6.3/lib/python3.10/site-packages/setuptools/build_meta.py", line 484, in run_setup
fastparquet>       super(_BuildMetaLegacyBackend,
fastparquet>     File "/nix/store/0d7f64w6zqxvn411gas3bqvpcik84x82-python3.10-setuptools-65.6.3/lib/python3.10/site-packages/setuptools/build_meta.py", line 335, in run_setup
fastparquet>       exec(code, locals())
fastparquet>     File "<string>", line 47, in <module>
fastparquet>     File "/nix/store/abax98471z8fshv4b9p46bkh3lxmpy0z-python3-3.10.9/lib/python3.10/subprocess.py", line 345, in call
fastparquet>       with Popen(*popenargs, **kwargs) as p:
fastparquet>     File "/nix/store/abax98471z8fshv4b9p46bkh3lxmpy0z-python3-3.10.9/lib/python3.10/subprocess.py", line 971, in __init__
fastparquet>       self._execute_child(args, executable, preexec_fn, close_fds,
fastparquet>     File "/nix/store/abax98471z8fshv4b9p46bkh3lxmpy0z-python3-3.10.9/lib/python3.10/subprocess.py", line 1847, in _execute_child
fastparquet>       raise child_exception_type(errno_num, err_msg, err_filename)
fastparquet>   FileNotFoundError: [Errno 2] No such file or directory: 'git'
fastparquet>   error: subprocess-exited-with-error
fastparquet>   
fastparquet>   × Preparing metadata (pyproject.toml) did not run successfully.
fastparquet>   │ exit code: 1
fastparquet>   ╰─> See above for output.
fastparquet>   
fastparquet>   note: This error originates from a subprocess, and is likely not a problem with pip.
fastparquet>   full command: /nix/store/abax98471z8fshv4b9p46bkh3lxmpy0z-python3-3.10.9/bin/python3.10 /nix/store/bk06jpp1hswx6i8zyzbk4lgkfyq92qic-python3.10-pip-22.3.1/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py prepare_metadata_for_build_wheel /build/tmpev3k0be0
fastparquet>   cwd: /build/source
fastparquet>   Preparing metadata (pyproject.toml) ... error
fastparquet> error: metadata-generation-failed
fastparquet> 
fastparquet> × Encountered error while generating package metadata.
fastparquet> ╰─> See above for output.
fastparquet> 
fastparquet> note: This is an issue with the package mentioned above, not pip.
fastparquet> hint: See above for details.